src/content/docs/en/reference/directives-reference.mdx

```diff
@@ -141,6 +143,17 @@ Load and hydrate the component JavaScript once the component has entered the use
 <HeavyImageCarousel client:visible />
 ```
 
+#### `client:visible={{rootMargin}}`
+<p><Since v="4.1.0" /></p>
+
+Optionally, pass a value to specify options to pass to the underlying `IntersectionObserver`. Currently, the only option supported is `rootMargin`. When `rootMargin` is specified, the component JavaScript will hydrate when a specified margin (in pixels) around the component enters the viewport, rather than the component itself.
```

Comment:
```suggestion
Optionally, a value for `rootMargin` can be passed to the underlying `IntersectionObserver`. When `rootMargin` is specified, the component JavaScript will hydrate when a specified margin (in pixels) around the component enters the viewport, rather than the component itself.
```